public/controllers/agent/components/register-agent.js

```diff
@@ -367,6 +367,14 @@ export const RegisterAgent = withErrorBoundary(
 
     agentNameVariable() {
       let agentName = `WAZUH_AGENT_NAME='${this.state.agentName}' `;
+      if (
+        this.state.selectedOS === 'macos' &&
+        this.state.selectedArchitecture &&
+        this.state.agentName !== ''
+      ) {
+        return agentName.replace(/=/g, ' ');
```

Comment:
**thought**: I dislike this solution, replacing the character. We should build the variable declaration for the different use cases and not build for one of them and then change it for a particular case. 

I guess we could improve it with a possible refactoring of the entire agent deployment guide.